pykube/objects.py

```diff
@@ -292,6 +300,20 @@ def cordon(self):
     def uncordon(self):
         self.unschedulable = False
 
+    def drain(self):
+        node_ip = self.ip
+        namespaces = Namespace.objects(self.api)
+        for namespace in namespaces:
+            pods = Pod.objects(self.api, namespace=namespace.name).all()
+            for pod in pods:
+                annotations = pod.obj["metadata"]["annotations"]
+                if annotations and annotations.get("kubernetes.io/created-by"):
+                    created_by = json.loads(annotations.get("kubernetes.io/created-by"))
+                    pod_type = created_by["reference"]["kind"]
+                    host_ip = pod.obj["status"].get("hostIP")
+                    if host_ip == node_ip and pod_type != "DaemonSet":
```

Comment:
Those get deleted with the kubectl drain iirc. I was trying to emulate that behavior as closely as possible